File: LeetCode Solutions/3. Longest Substring Without Repeating Characters/solution3.ts
function lengthOfLongestSubstring(s: string): number {
	let set = new Set(); let [head, tail, count, high] = [0, 0, 0, 0];
	while (head < s.length) {
		while(set.has(s[head])) {
			set.delete(s[tail++])
		}
		set.add(s[head++])
		count = head-tail
		if (count > high) high = count
	}
	return high
};

File: LeetCode Solutions/3. Longest Substring Without Repeating Characters/solution3.go
func lengthOfLongestSubstring(s string) int {
	hash := make(map[byte]bool)
	var head, tail, high int = 0, 0, 0
	
	for (head < len(s)){
		for (hash[s[head]]) {
			hash[s[tail]] = false; tail++
		}
		hash[s[head]] = true; head++
		if (head-tail > high) {high = head-tail}
	}
	return high
}